File: ios/Freedom/Core/Data/ConfigData.swift
// ═════════════════════════════════════════════════════════════════════════════
//  ConfigData — App configuration entity
// ═════════════════════════════════════════════════════════════════════════════

import Foundation
import GRDB

struct ConfigData: Codable, FetchableRecord, MutablePersistableRecord, Identifiable, Equatable {
    static let databaseTableName = "config_data"

    var id: Int64?
    var averageTripDistance: String = ""
    var averageTripsMonth: String = ""
    var monthlyBusinessKm: Double = 0.0
    var monthlyPrivateKm: Double = 0.0

    enum CodingKeys: String, CodingKey {
        case id
        case averageTripDistance = "average_trip_distance"
        case averageTripsMonth = "average_trips_month"
        case monthlyBusinessKm = "monthly_business_km"
        case monthlyPrivateKm = "monthly_private_km"
    }

    mutating func didInsert(_ inserted: InsertionSuccess) {
        id = inserted.rowID
    }
}

File: main/src/ui/java/freedom/app/data/entity/ConfigData.kt
package freedom.app.data.entity

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.PrimaryKey

@Entity(tableName = "config_data")
data class ConfigData(

    @PrimaryKey(autoGenerate = true)
    @ColumnInfo(name = "id")
    var Id: Long,
    @ColumnInfo(name = "average_trips_distance")
    var averageTripDistance: String?,
    @ColumnInfo(name = "average_trips_month")
    var averageTripsMonth: String?,
    @ColumnInfo(name = "monthly_business_km")
    var monthlyBusinessKm: Double? = null,
    @ColumnInfo(name = "monthly_private_km")
    var monthlyPrivateKm: Double? = null,
)